q2_feature_classifier/_vsearch.py

```diff
@@ -25,53 +30,159 @@ def classify_consensus_vsearch(query: DNAFASTAFormat,
+                                    strand='both',
+                                    min_consensus=0.51,
+                                    reads_per_batch=0,
+                                    confidence=0.7,
+                                    read_orientation=None,
+                                    threads=1,
+                                    prefilter=True):
+    exclude = ctx.get_action('quality_control', 'exclude_seqs')
+    ccv = ctx.get_action('feature_classifier', 'classify_consensus_vsearch')
+    cs = ctx.get_action('feature_classifier', 'classify_sklearn')
+    filter_seqs = ctx.get_action('taxa', 'filter_seqs')
+    merge = ctx.get_action('feature_table', 'merge_taxa')
+
+    # randomly subsample reference sequences for rough positive filter
+    if prefilter:
+        ref = str(reference_reads.view(DNAFASTAFormat))
+        with tempfile.NamedTemporaryFile() as output:
+            cmd = ['vsearch', '--fastx_subsample', ref,
+                   '--sample_size', '100', '--fastaout', output.name]
+            _run_command(cmd)
+            sparse_reference = qiime2.Artifact.import_data(
+                'FeatureData[Sequence]', output.name)
+
+            # perform rough positive filter on query sequences
+            query, misses, = exclude(
+                query_sequences=query, reference_sequences=sparse_reference,
+                method='vsearch', perc_identity=perc_identity,
+                perc_query_aligned=query_cov, threads=threads)
+
+    # find exact matches, perform LCA consensus classification
+    taxa1, = ccv(query=query, reference_reads=reference_reads,
+                 reference_taxonomy=reference_taxonomy, maxaccepts=maxaccepts,
+                 strand=strand, min_consensus=min_consensus,
+                 search_exact=True, threads=threads)
+
+    # Annotate taxonomic assignments with classification method
+    taxa1 = _annotate_method(taxa1, 'VSEARCH')
+
+    # perform second pass classification on unassigned taxa
+    # filter out unassigned seqs
+    try:
+        query, = filter_seqs(sequences=query, taxonomy=taxa1,
+                             include=_get_default_unassignable_label())
+    except ValueError:
+        # get ValueError if all sequences are filtered out.
+        # so if no sequences are unassigned, return exact match results
+        return taxa1
+
+    # classify with sklearn classifier
+    taxa2, = cs(reads=query, classifier=classifier,
+                reads_per_batch=reads_per_batch, n_jobs=threads,
+                confidence=confidence, read_orientation=read_orientation)
+
+    # Annotate taxonomic assignments with classification method
+    taxa2 = _annotate_method(taxa2, 'sklearn')
+
+    # merge into one big happy result
+    taxa, = merge(data=[taxa2, taxa1])
+    return taxa
+
+
+output_descriptions = {
+    'classification': 'The resulting taxonomy classifications.'}
+
+parameters = {'maxaccepts': Int % Range(0, None),
```

Comment:
What do you think about using something like this, instead?

```suggestion
parameters = {'maxaccepts': Int % Range(1, None) | Str % Choices(['all']),
```
I imagine you would need to accommodate this in the pipeline, but, I think this is a clearer API for users.